File: 프로그래머스/lv2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 1
    stack = []
    
    for item in s:        
        if not stack:
            stack.append(item)
        else:
            if stack[-1] == item:
                stack.pop()
            else:
                stack.append(item)
    
    if not stack:
        return 1
    else:
        return 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.swift
import Foundation

func solution(_ s:String) -> Int{
    var s = s
    
    while true {
        var stack = [Character]()
        for c in s {
             if let last = stack.last, last == c {
                stack.removeLast()
            } else {
                stack.append(c)
            }
        }
        
        if stack.isEmpty {
            return 1
        } else if s.count == stack.count {
            return 0
        } else {
            s = stack.map{String($0)}.joined(separator: "")
        }
        
    }
    
    return 0
}